runtime/src/main/java/org/corfudb/util/NetworkUtils.java

```diff
@@ -18,6 +23,25 @@
 @Slf4j
 public class NetworkUtils {
 
+
+    /**
+     * Get all interfaces, including virtual interfaces.
+     * @return A list of all interfaces
+     * @throws SocketException
+     */
+    private static List<NetworkInterface> getAllInterfaces() throws SocketException {
+        List<NetworkInterface> allInterfaces = new ArrayList<>(50);
+        Stack<NetworkInterface> stack = new Stack<>();
+        stack.addAll(Collections.list(NetworkInterface.getNetworkInterfaces()));
+
+        while (!stack.isEmpty()) {
```

Comment:
This is a very strange way to express iterating over a list and collecting it's subinterfaces (why are you copying the enumeration use a stack (?)). Why don't you just write a loop:
```java
            List<NetworkInterface> l = new ArrayList<>();
            Enumeration<NetworkInterface> e = NetworkInterface.getNetworkInterfaces();
            while (e.hasMoreElements()) {
                NetworkInterface n = e.nextElement();
                l.add(n);
                l.addAll(Collections.list(n.getSubInterfaces()));
            }
```

 or you could express this functionally using streams:

```java
            Collections.list(NetworkInterface.getNetworkInterfaces())
                    .stream()
                    .flatMap(m -> {
                        List<NetworkInterface> l = Collections.list(m.getSubInterfaces());
                        l.add(m);
                        return l.stream(); })
                    .collect(Collectors.toList());
```

Reply:
I thought that that virtual addresses can be nested beyond two levels (i.e. sub interfaces can have other sub interfaces), but it turns out that is false. Will removal the stack traversal.

Reply:
done.